### ISSUE TITLE
Missing tiles in mapbox-gl when color filter is applied

### ISSUE THREAD

**Issue by jaskiratr** [klokantech/tileserver-gl#111]
Hi,
I've been using tileserver-gl lately and it works really well right out of box, until I try to extrude buildings with color filter applied to fill-extrusions in mapbox. Some of the tiles do not render and then it stops showing any tiles at all when I pan around.

![image](https://cloud.githubusercontent.com/assets/8619877/22235031/c08f8a18-e1ca-11e6-9281-028c34ba1563.png)

![image](https://cloud.githubusercontent.com/assets/8619877/22235318/d8ae193c-e1cc-11e6-8556-ae15cf74e919.png)

![image](https://cloud.githubusercontent.com/assets/8619877/22235224/372d40f6-e1cc-11e6-8e85-7b390543c420.png)

It doesn't seem to be a problem with Mapbox-gl as it can render the geometries with same filter on tileset server by mapbox studio.

![image](https://cloud.githubusercontent.com/assets/8619877/22235026/b8a486c8-e1ca-11e6-9a81-746666f3eaa6.png)

The offline tiles do render correctly when there is no color filter on height applied. 

![image](https://cloud.githubusercontent.com/assets/8619877/22235242/60a99bdc-e1cc-11e6-9c9f-905149e11e11.png)

I've been struggling for a while now, not sure if this is an appropriate place to share this question. Any help would be wonderful. Thanks!!

**Comment by petrsloup** [comment]
I don't actually see how this could be related to the server-side. However, the latest release has updated version of the javascript viewer (mapbox-gl-js v0.31.0) and it might solve the issue.

**Comment by jaskiratr** [comment]
That's whats been puzzling me as well. I wouldn't have thought this could be server side issue, till I checked that I was able to render tiles from Mapbox Studio Styles but not tileserver-gl. I am using mapbox-gl.js v0.31.0. The tiles are also under 500k size. Maybe I'll try dropping random features from the tileset and see if that helps. It might have to do something with tile specification. Let see...
Thanks for the advice!
